NIFI-13590 Refactor components in standard-processors bundle using current API methods

EndzeitBegins opened the pull request:
<!-- Licensed to the Apache Software Foundation (ASF) under one or more -->
<!-- contributor license agreements.  See the NOTICE file distributed with -->
<!-- this work for additional information regarding copyright ownership. -->
<!-- The ASF licenses this file to You under the Apache License, Version 2.0 -->
<!-- (the "License"); you may not use this file except in compliance with -->
<!-- the License.  You may obtain a copy of the License at -->
<!--     http://www.apache.org/licenses/LICENSE-2.0 -->
<!-- Unless required by applicable law or agreed to in writing, software -->
<!-- distributed under the License is distributed on an "AS IS" BASIS, -->
<!-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. -->
<!-- See the License for the specific language governing permissions and -->
<!-- limitations under the License. -->

# Summary

[NIFI-13590](https://issues.apache.org/jira/browse/NIFI-13590)

Makes use of the new APIs introduced for working with AllowableValue, e.g. in [NIFI-12573](https://issues.apache.org/jira/browse/NIFI-12573) or [NIFI-12452](https://issues.apache.org/jira/browse/NIFI-12452).
Makes use of newer Java APIs for declaring relationships and properties.
Applies some minor improvements recommended by IntelliJ available in newer Java versions.

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 21

### Licensing

- [x] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [x] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [x] Documentation formatting appears as expected in rendered files

exceptionfactory commented on nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GenerateTableFetch.java:
```diff
@@ -172,24 +171,11 @@ public class GenerateTableFetch extends AbstractDatabaseFetchProcessor {
-        pds.add(CUSTOM_ORDERBY_COLUMN);
-        pds.add(OUTPUT_EMPTY_FLOWFILE_ON_ZERO_RESULTS);
-        propDescriptors = Collections.unmodifiableList(pds);
+        propDescriptors = List.of(
```
This can be moved to a static variable.

exceptionfactory commented on nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GetFTP.java:
```diff
@@ -57,44 +55,21 @@
-        this.properties = Collections.unmodifiableList(properties);
-    }
+    private static final List<PropertyDescriptor> PROPERTIES = List.of(
+            FTPTransfer.HOSTNAME, FTPTransfer.PORT, FTPTransfer.USERNAME, FTPTransfer.PASSWORD,
+            FTPTransfer.CONNECTION_MODE, FTPTransfer.TRANSFER_MODE, FTPTransfer.REMOTE_PATH,
+            FTPTransfer.FILE_FILTER_REGEX, FTPTransfer.PATH_FILTER_REGEX, FTPTransfer.POLLING_INTERVAL,
+            FTPTransfer.RECURSIVE_SEARCH, FTPTransfer.FOLLOW_SYMLINK, FTPTransfer.IGNORE_DOTTED_FILES,
+            FTPTransfer.DELETE_ORIGINAL, FTPTransfer.CONNECTION_TIMEOUT, FTPTransfer.DATA_TIMEOUT,
+            FTPTransfer.MAX_SELECTS, FTPTransfer.REMOTE_POLL_BATCH_SIZE, FTPTransfer.USE_NATURAL_ORDERING,
+            FTPTransfer.PROXY_CONFIGURATION_SERVICE, FTPTransfer.PROXY_TYPE, FTPTransfer.PROXY_HOST,
+            FTPTransfer.PROXY_PORT, FTPTransfer.HTTP_PROXY_USERNAME, FTPTransfer.HTTP_PROXY_PASSWORD,
+            FTPTransfer.BUFFER_SIZE, FTPTransfer.UTF8_ENCODING
```
Please reformat with one property per line.

exceptionfactory commented on nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GetFile.java:
```diff
@@ -190,6 +188,11 @@ public class GetFile extends AbstractProcessor {
             .build();
 
+    private static final List<PropertyDescriptor> PROPERTIES = List.of(
+            DIRECTORY, FILE_FILTER, PATH_FILTER, BATCH_SIZE, KEEP_SOURCE_FILE, RECURSE, POLLING_INTERVAL,
+            IGNORE_HIDDEN_FILES, MIN_AGE, MAX_AGE, MIN_SIZE, MAX_SIZE
```
Please reformat with one property per line.

exceptionfactory commented on nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GetSFTP.java:
```diff
@@ -56,50 +54,23 @@
-        this.properties = Collections.unmodifiableList(properties);
-    }
+    private static final List<PropertyDescriptor> PROPERTIES = List.of(
+            SFTPTransfer.HOSTNAME, SFTPTransfer.PORT, SFTPTransfer.USERNAME, SFTPTransfer.PASSWORD,
+            SFTPTransfer.PRIVATE_KEY_PATH, SFTPTransfer.PRIVATE_KEY_PASSPHRASE, SFTPTransfer.REMOTE_PATH,
+            SFTPTransfer.FILE_FILTER_REGEX, SFTPTransfer.PATH_FILTER_REGEX, SFTPTransfer.POLLING_INTERVAL,
+            SFTPTransfer.RECURSIVE_SEARCH, SFTPTransfer.FOLLOW_SYMLINK, SFTPTransfer.IGNORE_DOTTED_FILES,
+            SFTPTransfer.DELETE_ORIGINAL, SFTPTransfer.CONNECTION_TIMEOUT, SFTPTransfer.DATA_TIMEOUT,
+            SFTPTransfer.HOST_KEY_FILE, SFTPTransfer.MAX_SELECTS, SFTPTransfer.REMOTE_POLL_BATCH_SIZE,
+            SFTPTransfer.STRICT_HOST_KEY_CHECKING, SFTPTransfer.USE_KEEPALIVE_ON_TIMEOUT, SFTPTransfer.USE_COMPRESSION,
+            SFTPTransfer.USE_NATURAL_ORDERING, SFTPTransfer.PROXY_CONFIGURATION_SERVICE, FTPTransfer.PROXY_TYPE,
+            FTPTransfer.PROXY_HOST, FTPTransfer.PROXY_PORT, FTPTransfer.HTTP_PROXY_USERNAME,
+            FTPTransfer.HTTP_PROXY_PASSWORD, SFTPTransfer.CIPHERS_ALLOWED, SFTPTransfer.KEY_ALGORITHMS_ALLOWED,
+            SFTPTransfer.KEY_EXCHANGE_ALGORITHMS_ALLOWED, SFTPTransfer.MESSAGE_AUTHENTICATION_CODES_ALLOWED
```
Please reformat with one property per line.

exceptionfactory commented on nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/HandleHttpRequest.java:
```diff
@@ -312,37 +311,20 @@ public class HandleHttpRequest extends AbstractProcessor {
             .build();
+
+    private static final List<PropertyDescriptor> PROPERTIES = List.of(
+            PORT, HOSTNAME, SSL_CONTEXT, HTTP_PROTOCOL_STRATEGY, HTTP_CONTEXT_MAP, PATH_REGEX, URL_CHARACTER_SET,
+            ALLOW_GET, ALLOW_POST, ALLOW_PUT, ALLOW_DELETE, ALLOW_HEAD, ALLOW_OPTIONS, MAXIMUM_THREADS,
+            ADDITIONAL_METHODS, CLIENT_AUTH, CONTAINER_QUEUE_SIZE, MULTIPART_REQUEST_MAX_SIZE,
+            MULTIPART_READ_BUFFER_SIZE, PARAMETERS_TO_ATTRIBUTES
```
Please reformat with one property per line.

exceptionfactory commented on nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/IdentifyMimeType.java:
```diff
@@ -134,13 +131,15 @@ public class IdentifyMimeType extends AbstractProcessor {
             .build();
 
+    private static final List<PropertyDescriptor> PROPERTIES =
+            List.of(USE_FILENAME_IN_DETECTION, CONFIG_STRATEGY, MIME_CONFIG_BODY, MIME_CONFIG_FILE);
```
Please reformat with one property per line.

exceptionfactory commented on nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/JoinEnrichment.java:
```diff
@@ -189,18 +186,10 @@ public class JoinEnrichment extends BinFiles {
-        TIMEOUT
-    ));
+    private static final List<PropertyDescriptor> PROPERTIES = List.of(
+            ORIGINAL_RECORD_READER, ENRICHMENT_RECORD_READER, RECORD_WRITER, JOIN_STRATEGY, SQL, DEFAULT_PRECISION,
+            DEFAULT_SCALE, INSERTION_RECORD_PATH, MAX_BIN_COUNT, TIMEOUT
```
Please reformat with one property per line.

exceptionfactory commented on nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/JoinEnrichment.java:
```diff
@@ -223,23 +212,18 @@ public class JoinEnrichment extends BinFiles {
-        REL_TIMEOUT,
-        REL_FAILURE
-    )));
+    private static final Set<Relationship> RELATIONSHIPS = Set.of(REL_JOINED, REL_ORIGINAL, REL_TIMEOUT, REL_FAILURE);
```
Please reformat with one property per line.

exceptionfactory reviewed (changes requested):
Thanks for working on these improvements @EndzeitBegins. The changes look good in general, but the main recommendation is to keep one property descriptor and one relationship per line. This makes the list much easier to read, and easier to maintain when introducing changes. I noted a number of specific places for this, but did not note it on every Processor, although it applies to the majority of the changes.

EndzeitBegins commented:
Thank you for the quick review @exceptionfactory. 

I've adjusted the property declarations accordingly.

EndzeitBegins commented:
I've moved some more property declarations to static fields and removed the `PROP_` prefix of serval properties as you've suggested @exceptionfactory.
I'd appreciate another review.
